src/backend/cuda/kernel/scan_dim.hpp

```diff
@@ -128,7 +128,7 @@ namespace kernel
             }
     }
 
-    template<typename To, af_op_t op, int dim>
+    template<typename To, af_op_t op, int dim, bool inclusive_scan>
```

Comment:
I think this argument can be a parameter instead of a template argument. There isn't anything that is going on that would significantly change based on this parameter.